designs/resource-properties.md

```diff
@@ -0,0 +1,490 @@
+# Resource properties
```

Comment:
Didn't we add member elision support for resource properties? And also support for the `for` keyword in structure definitions:

```
structure Foo for MyResource {
    $baz
}

Reply:
Added support in SetResourceBasedTargets.java and unit test to verify it's working.